pkg/virt-controller/watch/vm.go

```diff
@@ -929,25 +1041,84 @@ func (c *VMController) updateStatus(vm *virtv1.VirtualMachine, vmi *virtv1.Virtu
 	createdMatch := created == vm.Status.Created
 
 	ready := false
+
 	if created {
 		ready = controller.NewVirtualMachineInstanceConditionManager().HasConditionWithStatus(vmi, virtv1.VirtualMachineInstanceConditionType(k8score.PodReady), k8score.ConditionTrue)
 	}
 	readyMatch := ready == vm.Status.Ready
 
-	if errMatch && createdMatch && readyMatch {
+	runStrategy, err := vm.RunStrategy()
+	if err != nil {
+		log.Log.Object(vm).Errorf("Error getting RunStrategy: %v", err)
+	}
+	clearChangeRequest := false
+	if len(vm.Status.StateChangeRequests) != 0 {
+		// Only consider one stateChangeRequest at a time. The second and subsequent change
+		// requests have not been acted upon by this controller yet!
+		stateChange := vm.Status.StateChangeRequests[0]
+		if stateChange.Action == virtv1.StopRequest {
+			if vmi == nil {
+				// because either the VM or VMI informers can trigger processing here
+				// double check the state of the cluster before taking action
+				_, err = c.clientset.VirtualMachine(vm.ObjectMeta.Namespace).Get(vm.GetName(), &v1.GetOptions{})
+				if err != nil && errors.IsNotFound(err) {
+					// If there's no VMI, then the VMI was stopped, and the stopRequest can be cleared
+					log.Log.Object(vm).V(4).Infof("No VMI. Clearing stop request")
+					clearChangeRequest = true
+				}
+			} else {
+				if stateChange.UID == nil {
+					// It never makes sense to have a request to stop a VMI that doesn't
+					// have a UUID associated with it. This shouldn't be possible -- but if
+					// it occurs, clear the stopRequest because it can't be acted upon
+					log.Log.Object(vm).Errorf("Stop Request has no UID.")
+					clearChangeRequest = true
+				} else if *stateChange.UID != vmi.UID {
+					// If there is a VMI, but the UID doesn't match, then it
+					// must have been previously stopped, so the stopRequest can be cleared
+					log.Log.Object(vm).V(4).Infof("VMI's UID doesn't match. clearing stop request")
+					clearChangeRequest = true
+				} else if vmi.IsFinal() {
```

Comment:
don't clear the stop case until the VMI has been deleted. IsFinal() is not the same as deleted.